Code/Framework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1314,4 +1410,243 @@ namespace UnitTest
+                // Ensuring that this node has the correct parent
+                ASSERT_EQ(child.GetParent()->GetClassMetadata()->m_name, "GroupTestComponent");
+            }
+        }
+    }
+
+    TEST_F(InstanceDataHierarchyGroupTestFixture, TestNestedGroups)
+    {
+        using namespace AzToolsFramework;
+
+        // Setting up the data node hierarchy
+        AZ::SerializeContext serializeContext;
+        serializeContext.CreateEditContext();
+        Entity::Reflect(&serializeContext);
+        GroupTestComponent::Reflect(&serializeContext);
+
+        AZStd::unique_ptr<AZ::Entity> testEntity1(new AZ::Entity());
+        testEntity1->CreateComponent<GroupTestComponent>();
+
+        InstanceDataHierarchy instanceDataHierarchy;
+        instanceDataHierarchy.AddRootInstance(testEntity1.get());
+        instanceDataHierarchy.Build(&serializeContext, 0);
+
+        // Adding the nodes to a node stack
+        auto rootNode = instanceDataHierarchy.GetRootNode();
+        AZStd::stack<InstanceDataNode*> nodeStack;
+        nodeStack.push(rootNode);
+        InstanceDataNode* componentNode1 = nullptr;
+        while (!nodeStack.empty())
+        {
+            InstanceDataNode* node = nodeStack.top();
+            nodeStack.pop();
+            if (node->GetClassMetadata()->m_typeId == AZ::AzTypeInfo<GroupTestComponent>::Uuid())
+            {
+                componentNode1 = node;
+                break;
+            }
+            for (InstanceDataNode& child : node->GetChildren())
+            {
+                nodeStack.push(&child);
+            }
+        }
+        // Iterating through the children in the instance data hierarchy to verify their properties
+        ASSERT_TRUE(componentNode1 != nullptr);
+        for (auto child : componentNode1->GetChildren())
+        {
+            AZStd::string childName(child.GetElementMetadata()->m_name);
+            if (childName.compare("SubDataNormal") == 0)
+            {
+                for (InstanceDataNode& subChild : child.GetChildren())
+                {
+                    childName = subChild.GetElementMetadata()->m_name;
+                    if (childName.compare("SubInt") == 0)
+                    {
+                        // False for any child node with serializable data
+                        ASSERT_FALSE(subChild.GetElementEditMetadata()->IsClassElement());
+                        // Child node should never be a ClassElement::Group, unless it is the root node of a ToggleGroup
+                        ASSERT_NE(subChild.GetElementEditMetadata()->m_elementId, AZ::Edit::ClassElements::Group);
+                        // Ensuring that this node was assigned to the appropriate group
+                        ASSERT_EQ(subChild.GetGroupElementMetadata()->m_description, "Normal SubGroup");
+                        // Ensuring that this node has the correct parent
+                        ASSERT_EQ(subChild.GetParent()->GetClassMetadata()->m_name, "SubData");
```

Comment:
suggestion: It might be possible to rewrite some of this using matchers too (or even a parameterized test which can remove a lot of boilerplate) - the tests are great but we want to try and treat them the same as we treat production code and make them as clear and compact as we can

Reply:
I implemented a parameterized test and changed the layout of the other tests to make them more compact and provide greater feedback when they fail :)

Reply:
That's fantastic @dtamkin1, thanks very much for the update! Looking forward to seeing them :+1: